editor/editor_node.cpp

```diff
@@ -1291,7 +1291,7 @@ void EditorNode::_dialog_action(String p_file) {
 			uint32_t current = editor_history.get_current();
 			Object *current_obj = current > 0 ? ObjectDB::get_instance(current) : NULL;
 
-			ERR_FAIL_COND(!current_obj->cast_to<Resource>())
```

Comment:
I will wait until #1058 gets merged, but using `->cast_to` on null pointers is already causing segfaults with GCC 7.1.1. I fixed one on #10492 and you can see the full backtrace I got on #10372 after the segfault.